### PR TITLE
LPS-197435 Migrate (some) selectors to typescript

### PR DESCRIPTION
**Pull request by veroglez**

### CODE REVIEW

**Comment by liferay-echo** [comment]
Thank you for submitting this PR. 👏

Make sure it follows the [Echo Pull Request Template](https://gist.github.com/liferay-echo/5a4455285ff1800f9da146816d85e67d#file-pull-request-template-md) if you haven't already done so. 

Learn more about it [here](https://liferay.atlassian.net/wiki/spaces/ENGECHO/pages/2027455014/Echo+Pull+Request+Etiquette).

File: modules/apps/layout/layout-content-page-editor-web/src/main/resources/META-INF/resources/page_editor/types/layout_data/BaseLayoutDataItem.d.ts [hunk 1]
```diff
@@ -45,8 +45,9 @@ export interface BaseLayoutDataItem<
 	Config
 > {
 	children: string[];
-	config: Config;
+	config: Config & {name?: string};
 	itemId: string;
 	itemType: ItemType;
 	parentId: string;
+	type: ItemType;
```
Comment (sandrodw3) [inline modules/apps/layout/layout-content-page-editor-web/src/main/resources/META-INF/resources/page_editor/types/layout_data/BaseLayoutDataItem.d.ts:53]: Now that I see this, do we have `itemType` in any `layoutData` item?

**Review by sandrodw3 — COMMENTED** [review]
Awesome 🤍 just a tiny question

**Comment by veroglez** [comment]
ci:forward

File: modules/apps/layout/layout-content-page-editor-web/src/main/resources/META-INF/resources/page_editor/types/layout_data/BaseLayoutDataItem.d.ts (hunk 1 above)
Comment (veroglez) [reply modules/apps/layout/layout-content-page-editor-web/src/main/resources/META-INF/resources/page_editor/types/layout_data/BaseLayoutDataItem.d.ts:52]: Yep, it's an error, no layoutData item has itemType. I've updated the code.

**Comment by Tim-Cao** [comment]
ci:forward:force

**Pull request closed without merging**